src/client/public/functions.ts

```diff
@@ -21,10 +23,20 @@ Office.initialize = () => {
         if (typeof options.displayInIframe === 'undefined') {
             options.displayInIframe = true;
         }
-
-        Office.context.ui.displayDialogAsync(url, options, null);
-
-        if (event) {
+        Office.context.ui.displayDialogAsync(url, options, (result) => {
+            if (Utilities.host === HostType.OUTLOOK) {
+                if (result.status === Office.AsyncResultStatus.Failed) {
+                    event.completed();
+                }
+                let dialog = result.value as Office.DialogHandler;
+                dialog.addEventHandler(Office.EventType.DialogEventReceived, () => {
+                    if (event) {
+                        event.completed();
+                    }
+                });
+            }
+        });
+        if (event && Utilities.host !== HostType.OUTLOOK) {
```

Comment:
Does having our "event.completed()" call inside the callback not work?  Why the different code paths?